extension/authoidcextension/extension.go

```diff
@@ -29,23 +29,28 @@ import (
 	"time"
 
 	"github.com/coreos/go-oidc"
+	"go.uber.org/zap"
 	"google.golang.org/grpc"
+
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/config/configauth"
 )
 
-type oidcAuthenticator struct {
-	attribute string
-	config    OIDC
-	provider  *oidc.Provider
-	verifier  *oidc.IDTokenVerifier
+type oidcExtension struct {
+	cfg               *Config
+	unaryInterceptor  configauth.UnaryInterceptorFunc
+	streamInterceptor configauth.StreamInterceptorFunc
```

Comment:
Are these needed for testing only or we override them in some other cases as well?

Reply:
They are for testing only, I don't think they would be useful in other contexts.